### PR TITLE
Merged with vlad main to eval on cpu; added copier and residual policy

### PR DESCRIPTION
**Pull request by maxwshen**
merged with vlad main

### CODE REVIEW

File: embed.py [hunk 1]
```diff
@@ -254,8 +254,8 @@ def sample(self, embedded):
 embed_action = EnumEmbedding(enums.Action)
 embed_char = EnumEmbedding(enums.Character)
 
-# 8 should be enough for puff and kirby
-embed_jumps_left = OneHotEmbedding("jumps_left", 8, dtype=np.uint8)
+# puff and kirby have 6 jumps
+embed_jumps_left = OneHotEmbedding("jumps_left", 6)
```
Comment (vladfi1) [inline embed.py:258]: Please keep the dtype=np.uint8, it's actually necessary later on.

File: policies.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import sonnet as snt
```
Comment (vladfi1) [inline policies.py:1]: Did you use `git mv` on this? I think if you do then the diff will show up properly.

Comment (vladfi1) [reply policies.py:1]: (Don't need to fix it now, just an fyi for the future.)

File: train.py [hunk 3]
```diff
@@ -42,15 +43,56 @@ def config():
   )
   learner = Learner.DEFAULT_CONFIG
   network = networks.DEFAULT_CONFIG
+  policy = policies.DEFAULT_CONFIG
   expt_dir = train_lib.get_experiment_directory()
 
+class TrainManager:
```
Comment (vladfi1) [inline train.py:49]: This was moved to a separate file, I think you accidentally kept it when you merged.

File: train.py [hunk 4]
```diff
@@ -154,4 +196,10 @@ def main(dataset, expt_dir, _config, _log):
     save_path = save()
     if save_path:
       _log.info('Saved network to %s', save_path)
+<<<<<<< HEAD
```
Comment (vladfi1) [inline train.py:199]: You have a merge conflict left here.

**Review by vladfi1 — CHANGES_REQUESTED** [review]
Overall looks good. Left some comments on things to be addressed.

File: policies.py (hunk 2 above)
Comment (maxwshen) [reply policies.py:1]: I didn't use `git mv`, just `git add policies.py` and `git rm policy.py`. Not sure on best practices, so definitely open to feedback

File: train.py [hunk 5]
```diff
@@ -196,10 +174,6 @@ def main(dataset, expt_dir, _config, _log):
     save_path = save()
     if save_path:
       _log.info('Saved network to %s', save_path)
-<<<<<<< HEAD
-    save_model()
-=======
 
 if __name__ == '__main__':
```
Comment (vladfi1) [inline train.py:178]: You merged the wrong way. It should have save_model() but not app.run(main).

**Review by vladfi1 — CHANGES_REQUESTED** [review]

**Review by vladfi1 — APPROVED** [review]

**Pull request merged**